include/boost/decimal/detail/cmath/exp.hpp

```diff
@@ -43,20 +43,13 @@ constexpr auto exp_impl(T x) noexcept
     {
         if (fpc == FP_INFINITE)
         {
-            if (signbit(x))
-            {
-                result = zero;
-            }
-            else
-            {
-                result = x;
-            }
+            result = (signbit(x) ? zero : std::numeric_limits<T>::infinity());
         }
         else if (fpc == FP_NAN)
         {
-            result = x;
+            result = std::numeric_limits<T>::quiet_NaN();
```

Comment:
I think this should stay as `result = x` so we return the correct type of NAN from the input.